targets/TARGET_Silicon_Labs/TARGET_EFM32/TARGET_EFM32GG11/device/system_efm32gg11b.h

```diff
@@ -39,21 +37,38 @@ extern "C" {
 
 #include <stdint.h>
 
-/**************************************************************************//**
+/***************************************************************************//**
  * @addtogroup Parts
  * @{
- *****************************************************************************/
-/**************************************************************************//**
+ ******************************************************************************/
+/***************************************************************************//**
  * @addtogroup EFM32GG11B EFM32GG11B
  * @{
- *****************************************************************************/
+ ******************************************************************************/
+
+/*******************************************************************************
+ ******************************   TYPEDEFS   ***********************************
+ ******************************************************************************/
+
+/* Interrupt vectortable entry */
+typedef union {
+  void (*pFunc)(void);
+  void *topOfStack;
+} tVectorEntry;
 
 /*******************************************************************************
  **************************   GLOBAL VARIABLES   *******************************
  ******************************************************************************/
 
-extern uint32_t SystemCoreClock;        /**< System Clock Frequency (Core Clock) */
-extern uint32_t SystemHfrcoFreq;        /**< System HFRCO frequency */
+extern uint32_t SystemCoreClock;    /**< System Clock Frequency (Core Clock) */
+extern uint32_t SystemHfrcoFreq;    /**< System HFRCO frequency */
+
+#if defined(__VTOR_PRESENT) && (__VTOR_PRESENT == 1U)
+#if defined(__ICCARM__)    /* IAR requires the __vector_table symbol */
+#define __Vectors    __vector_table
```

Comment:
this breaks the build (it is only for IAR so the failure looks valid)

What mbed boot for IAR expects: `extern void *__vector_table;`

Reply:
Is the type of __vector_table defined in a spec somewhere? If IAR only expects the symbol to exist, it seems wrong to make assumptions about its type in code...

Reply:
Good question. I quickly found one IAR reference manual that contains `const IntVector __vector_table[]`.

cc @ARMmbed/mbed-os-core Please review

Reply:
The variable is declared as extern in mbed_boot_iar.c
```
extern void *__vector_table;
```

It should not be declared again as 
```
extern const tVectorEntry __vector_table[];
```
But it should be defined.

Reply:
There are some mentions of this symbol on IAR pages, or some code on github. It is defined as `const IntVector __vector_table[] =` so we should use the same type `IntVector`.

I would update our boot file to use `extern const tVectorEntry __vector_table[];`

Reply:
I'll send PR shortly

Reply:
sent, https://github.com/ARMmbed/mbed-os/pull/12696